src/main/java/com/adobe/commerce/cif/api/ProductApi.java

```diff
@@ -60,6 +60,21 @@ Product getProductById(
         @HeaderParam(ACCEPT_LANGUAGE) String acceptLanguage
     );
 
+    @GET
+    @Path("/slug/{slug}")
+    @ApiOperation(value = "Returns a product by slug.")
+    @ApiResponses(value = {
+            @ApiResponse(code = HTTP_BAD_REQUEST, message = HTTP_BAD_REQUEST_MESSAGE, response = ErrorResponse.class),
+            @ApiResponse(code = HTTP_NOT_FOUND, message = HTTP_NOT_FOUND_MESSAGE, response = ErrorResponse.class)
+    })
+    Product getProductBySlug(
```

Comment:
I think that this is possible if we change the `byId()` to a search based on filters, we can then use the available `or` condition to search on id or on slug.

Reply:
@masodu your proposal might work for some commerce providers but main be not for all. Also changing to search might have performance impact over accessing objects directly via id / key.

Reply:
@mhaack I am not sure about the performance impact but if we really want we can do it (at least for CT and magento). Personally I don't have a very strong opinion but if we go with the current approach we might limit the sub resource REST style implementation ( if we get to the conclusion that we need it):  `product/slug/meskwielt.1-xs/variants` instead of `product/meskwielt.1-xs/variants`. I prefer option 2 for simplicity while I agree that option 1 can be more flexible for the implementation.

Reply:
@masodu your proposed change would also require us to do a major release, since the change is not backwards compatible in terms of semver.

Reply:
I don't see that need. We only need to change the documentation and use as id either the internal id or the slug.

Reply:
@masodu Can you please make a concrete example of your proposal?

Reply:
I disagree that `getProductById` could use either `id` or `slug`, this is going to be confusing and would unnecessarily make our implementation more complicated, and there could be cases where we need both (that is, be able to resolve by id and by slug).